app/Models/User.php

```diff
@@ -388,14 +388,37 @@ public static function checkWhenUsernameAvailable($username) : Carbon
 
     public function getUsernameAvailableAt() : Carbon
     {
-        if ($this->group_id !== 2 || $this->user_type === 1) {
+        $playCount = array_reduce(array_keys(Beatmap::MODES), function ($result, $mode) {
+            return $result + $this->statistics($mode, true)->value('playcount');
+        }, 0);
+
+        if ($this->group_id !== 2) {
             //reserved usernames
-            return Carbon::now()->addYears(10);
+            return Carbon::now()->addYears(10);  //This will always be in the future, which is wanted
         }
 
+        if ($this->user_type === 1) {
+            $minDays = 0;
+            $expMod = 0.35;
+            $linMod = 0.75;
+        } else {
+            $minDays = static::INACTIVE_DAYS;
+            $expMod = 1;
+            $linMod = 1;
+        }
+
+        // This is a exponential decay function with the identity 1-e^{-$playCount}.
+        // The constant multiplier of 1580 causes the formula to flatten out at around 1580 days (~4.3 years).
+        // $playCount is then divided by the constant value 5900 causing it to flatten out at about 40,000 plays.
+        // Furthermore, when the user is restricted, the formula has a 0.35 multiplier on $playCount,
+        // causing it to flatten out more slowly and a linear bonus of 0.75 to reward long-term players.
```

Comment:
The linear bonus should be mentioned in previous part instead and only mention the bonus is lower for restricted user here.